# Probability

In [8]:
import pandas as pd
from original import outcomes, cal_prob

## Probability for each case

In [9]:
probs = pd.DataFrame({cal_prob(case) for name, case in outcomes.items()}).set_index('case')

In [10]:
probs

prob  payoff       expR
case                             
840   0.000155     300   0.046595
831   0.000994     100   0.099403
651   0.027833      10   0.278327
822   0.000870     120   0.104373
732   0.027833      15   0.417491
444   0.126842       5   0.634209
633   0.097414       5   0.487072
660   0.000870     100   0.086977
552   0.097414       5   0.487072
750   0.000994      60   0.059642
543   0.487072     -30 -14.612175
741   0.009940      40   0.397610
642   0.121768       5   0.608841

## Probability grouped by same reward amount

In [11]:
probs_grouped = probs.groupby('payoff').sum()

In [12]:
probs_grouped

prob       expR
payoff                     
-30     0.487072 -14.612175
 5      0.443439   2.217195
 10     0.027833   0.278327
 15     0.027833   0.417491
 40     0.009940   0.397610
 60     0.000994   0.059642
 100    0.001864   0.186380
 120    0.000870   0.104373
 300    0.000155   0.046595

## Expected rewards

In [13]:
probs_grouped.expR.sum() - 20/10

-12.904563198277021

As you can see, on average the player is lossing 12 dollars on every play. Clearly this game is a scam.

This is because the player is having close to 49% of getting case `543` which corresponds to a punishment of $30.

Although in some cases player may win more $100 - $300, but these are extremely rare cases.

Without indepth calculation, the player may think that these event are equally likely to occure, but we have shown that this is far from being true.

It is this huge difference in perception which makes this game a scam.

## Sum of all probabilities

In [14]:
sum(probs_grouped.prob)

0.9999999999999999